admin/class-virgil-pure-admin.php

```diff
@@ -266,6 +266,21 @@ public function virgil_pure_profile_update(int $userId): void
         $wpUser = get_user_by('id', $userId);
         $this->updatePassword($wpUser);
     }
+    public function virgil_pure_user_register(int $userId): void
+    {
+        if (get_option(Option::AUTO_MIGRATION)) {
+            if ($this->dbqh->isQueueEmpty(get_option(Option::AUTO_MIGRATION))) {
+                $this->fh->migrate();
+                update_option(Option::LAST_CHECK, time());
+                return;
+            }
+
+            if (time() - get_option(Option::LAST_CHECK) > 300) {
+                $this->fh->migrate();
+                update_option(Option::LAST_CHECK, time());
+            }
+        }
+    }
```

Comment:
This hook is not needed.
Instead create a background job that will fetch non-migrated users each N minutes and migrate them.